api/src/client/ApiTaxFilingClient.test.ts

```diff
@@ -277,7 +277,14 @@ describe("ApiTaxFilingClient", () => {
         taxIdAndBusinessNameAndEmail.email,
         taxIdAndBusinessNameAndEmail.businessName
       );
-      expect(response).toEqual({ state: "FAILED", errorField: stubResponse.Errors[0].Field });
+      let errorField;
+      if (stubResponse.Errors[0].Field == "Taxpayer ID") {
+        errorField = "formFailure";
+      }
+      if (stubResponse.Errors[0].Field == "Business Name") {
+        errorField = "businessName";
+      }
```

Comment:
I think that what's going on here, if I am reading it correctly, is that we're generating a stub response with a random error field that has one of two values. And then here, we're setting up our expectation based on which of those two random values it is - if it's X, then we expect A, if it's Y, then we expect B.

If that's right - this kind of logic shouldn't really exist in a test. We're kind of just duplicating the logic of the code we're testing. If you want to test both of these cases (which I assume you do) then we should have 2 tests to cover both outcomes. In each test, set one value explicitly in the stub response, and then explicitly expect what you should receive back in the response based on that input.